### PR TITLE
chore(deps): bump phpunit to v10

### PR DESCRIPTION
**Pull request by simPod**

### CODE REVIEW

**Comment by simPod** [comment]
Also, I'd like to ask for a new release with newer infection support if possible (https://github.com/infection/infection/releases/tag/0.26.19)

File: phpunit.xml.dist [hunk 1]
```diff
@@ -3,26 +3,23 @@
...
     xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd"
     bootstrap="./vendor/autoload.php"
-    beStrictAboutCoversAnnotation="true"
-    beStrictAboutResourceUsageDuringSmallTests="true"
     beStrictAboutChangesToGlobalState="true"
+    beStrictAboutCoverageMetadata="true"
     beStrictAboutOutputDuringTests="true"
-    beStrictAboutTodoAnnotatedTests="true"
     executionOrder="random"
+    cacheDirectory=".phpunit.cache"
```
Comment (Ocramius) [inline phpunit.xml.dist]: Should be skipped

**Review by Ocramius — COMMENTED** [review]
Are coverage annotations not to be migrated to attributes? 🤔

**Comment by simPod** [comment]
`beStrictAboutCoversAnnotation` is renamed to `beStrictAboutCoverageMetadata`

`beStrictAboutResourceUsageDuringSmallTests` and `beStrictAboutTodoAnnotatedTests` are removed completely

File: phpunit.xml.dist (hunk 1 above)
Comment (simPod) [reply phpunit.xml.dist]: Do we want to keep `.phpunit.result.cache` files?

Comment (Ocramius) [reply phpunit.xml.dist]: Nope, we don't really need any caching at all :thinking:

**Review by Ocramius — APPROVED** [review]
Thanks @simPod!

**Comment by simPod** [comment]
Ok but there's still caching enabled via `cacheResult` which is default `true`. I will revert gitignore change.

**Comment by simPod** [comment]
Done. That's it then.

**Comment by Ocramius** [comment]
Thanks @simPod!

**Pull request merged**